app/src/main/java/com/amaze/filemanager/filesystem/files/FileUtils.java

```diff
@@ -416,6 +417,18 @@ public static float readableFileSizeFloat(long size) {
    */
   public static void installApk(
       final @NonNull File f, final @NonNull PermissionsActivity permissionsActivity) {
+
+    try {
+      PackageInstallValidation.validatePackageInstallability(f);
+    } catch (IllegalArgumentException e) {
```

Comment:
Did you mean `PackageCannotBeInstalledException`? When would the `Toast` get fired? Trying to install Amaze crashes the app for me.